src/workerd/jsg/jsg.h

```diff
@@ -2063,6 +2107,68 @@ class Lock {
   ContextScope enterContextScope(v8::Local<v8::Context> context);
   // Ensures that we are currently within the scope of the given v8::Context
 
+  // ====================================================================================
+  JsValue undefined() KJ_WARN_UNUSED_RESULT;
+  JsValue null() KJ_WARN_UNUSED_RESULT;
+  JsBoolean boolean(bool val) KJ_WARN_UNUSED_RESULT;
+  JsNumber num(double) KJ_WARN_UNUSED_RESULT;
+  JsNumber num(float) KJ_WARN_UNUSED_RESULT;
+  JsInt32 num(int8_t) KJ_WARN_UNUSED_RESULT;
+  JsInt32 num(int16_t) KJ_WARN_UNUSED_RESULT;
+  JsInt32 num(int32_t) KJ_WARN_UNUSED_RESULT;
+  JsBigInt num(int64_t) KJ_WARN_UNUSED_RESULT;
+  JsUint32 num(uint8_t) KJ_WARN_UNUSED_RESULT;
+  JsUint32 num(uint16_t) KJ_WARN_UNUSED_RESULT;
+  JsUint32 num(uint32_t) KJ_WARN_UNUSED_RESULT;
+  JsBigInt num(uint64_t) KJ_WARN_UNUSED_RESULT;
+  JsString str() KJ_WARN_UNUSED_RESULT;
+  template <typename T>
+  JsString str(kj::ArrayPtr<T>) KJ_WARN_UNUSED_RESULT;
```

Comment:
This confused me for a while, I thought this was offering to stringify arbitrary arrays using KJ stringification, but didn't support arbitrary values outside of arrays.

But I see now `v8Str` is templated over `ArrayPtr<T>` but is actually implemented as a series of `if constexpr (isSameType<T, whatever>())` handling each possible type... with KJ_UNREACHABLE as a fallback.

IMO `v8Str` should just have overloads for each supported type -- it would be easier to read and about the same amount of code anyway.

If not that, then at the very least, the KJ_UNREACHABLE should be replaced with a static_assert so that it catches unsupported types at compile time instead of runtime.

And here, we should again either have explicit overloads or at least a comment saying what's allowed.

Reply:
Yeah, I was thinking the same (having specific overrides rather than the `if constexpr` block. It would be much nicer for many reasons. I'll switch it over to that here for `js.str(...)` but I don't want to go through the hassle of changing `jsg::v8Str(...)` right now if we're just going to switch things over to `js.str(...)` soon.

Reply:
Updated